packages/input-container/src/vaadin-input-container-styles.js

```diff
@@ -68,7 +72,27 @@ export const inputContainerStyles = css`
     /* because ::slotted(...)::placeholder does not work in Safari. */
     font: inherit;
     color: inherit;
-    /* Override default opacity in Firefox */
-    opacity: 1;
+  }
+
+  ::slotted(input:placeholder-shown) {
+    color: var(--vaadin-input-field-placeholder-color, var(--_vaadin-color));
+  }
+
+  :host(:focus-within) {
+    outline: var(--vaadin-focus-ring-width) solid var(--vaadin-focus-ring-color);
+    outline-offset: calc(var(--vaadin-input-field-border-width, 1px) * -1);
+  }
+
+  :host([readonly]) {
+    border-style: dashed;
+  }
+
+  :host([disabled]) {
```

Comment:
I guess we could move `pointer-events: none;` here as well?

https://github.com/vaadin/web-components/blob/3399e37512975db5f5e8a045d3af5a341db67d34/packages/vaadin-lumo-styles/mixins/input-field-shared.js#L107-L108

Reply:
Maybe? I think the main reason for that style is to prevent `:hover` and `:active` styles from getting applied. But doesn't that also disable tooltips?